crates/lox-io/src/ndm/common.rs

```diff
@@ -0,0 +1,1624 @@
+pub struct LonType {
+    #[serde(rename = "$text")]
+    pub base: LonRange,
+    #[serde(rename = "@units")]
+    pub units: LatLonUnits,
+}
+
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    serde::Deserialize,
+    serde::Serialize,
+    lox_derive::KvnDeserialize,
+)]
+#[serde(default)]
+pub struct UserDefinedType {
+    #[serde(rename = "COMMENT")]
+    pub comment_list: Vec<String>,
+    #[serde(rename = "USER_DEFINED")]
+    pub user_defined_list: Vec<UserDefinedParameterType>,
+}
+
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    serde::Deserialize,
+    serde::Serialize,
+    lox_derive::KvnDeserialize,
+)]
+#[serde(default)]
+pub struct UserDefinedParameterType {
+    #[serde(rename = "$text")]
+    pub base: String,
+    #[serde(rename = "@parameter")]
+    pub parameter: String,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct QuaternionType {}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct QuaternionRateType {}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct QuaternionDotType {
+    #[serde(rename = "$text")]
+    pub base: f64,
+    #[serde(rename = "@units")]
+    pub units: Option<QuaternionDotUnits>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct RotationAngleType {
+    #[serde(rename = "rotation1")]
+    pub rotation1: RotationAngleComponentType,
+    #[serde(rename = "rotation2")]
+    pub rotation2: RotationAngleComponentType,
+    #[serde(rename = "rotation3")]
+    pub rotation3: RotationAngleComponentType,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct RotationAngleComponentTypeold {
+    #[serde(rename = "@units")]
+    pub units: Option<AngleUnits>,
+    #[serde(rename = "@angle")]
+    pub angle: AngleKeywordType,
+    #[serde(rename = "@value")]
+    pub value: f64,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct RotationAngleComponentType {
+    #[serde(rename = "$text")]
+    pub base: f64,
+    #[serde(rename = "@angle")]
+    pub angle: AngleKeywordType,
+    #[serde(rename = "@units")]
+    pub units: Option<AngleUnits>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct RotationRateType {
+    #[serde(rename = "rotation1")]
+    pub rotation1: RotationRateComponentType,
+    #[serde(rename = "rotation2")]
+    pub rotation2: RotationRateComponentType,
+    #[serde(rename = "rotation3")]
+    pub rotation3: RotationRateComponentType,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct RotationRateComponentTypeOld {
```

Comment:
Same question about whether the `Old`-suffixed types are still needed?

Reply:
Same as above.